ipld/amt/src/root.rs

```diff
@@ -0,0 +1,72 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use crate::Node;
+use encoding::{
+    de::{self, Deserialize},
+    ser::{self, Serialize},
+};
+
+/// Root of an AMT vector, can be serialized and keeps track of height and count
+#[derive(PartialEq, Debug)]
+pub(super) struct Root<V> {
+    pub(super) height: u32,
```

Comment:
Just a thought. But these fields can be `pub` instead of `pub(super)` since your struct is `pub(super)`, right? Doesn't matter though, just a though :~)

Reply:
This is more explicit for if the Root visibility changes in future. All of this is internal so no need for that <|:)